apps/aecore/src/aec_peers.erl

```diff
@@ -658,6 +667,13 @@ connected_peers(outbound, #state{ conns = Conns }) ->
     [ peer_info(P) || #conn{ peer = P, type = outbound, state = connected }
                       <- maps:values(Conns) ].
 
+-spec random_connected_peers(N :: pos_integer(), all | inbound | outbound, state())
+    -> [peer_info()].
+random_connected_peers(N, Tag, State) ->
```

Comment:
this is introduced for the the usage below in aec_sync. Don't we need in this very case only Outbound peers?

Reply:
From what @hanssv says, It seems there is some cases where nodes don't have outbound connection.

Reply:
So then we can't do anything. The point of change is to limit initial dataset which we use for picking up random nodes to the ones that are outbound?

Reply:
The first point of this PR is to send gossip to connected peers. Then the idea is to send them only to outbound. Theoretically, all the nodes should reach close to the maximum number of outbound connections (10), but @hanssv said he saw some nodes without outbound connection. We should look at these nodes and investigate why they don't have outbound connections and then it will be safe to only gossip to outbound connections.

Reply:
The cases without outbound connections are artificial I think, but for local testing networks it will definitely happen. Gossiping to only 10 nodes is low though...

Reply:
The limit could be raised or we could do a smarter connection selection that takes all the outbound connections plus top up to a configured number with random inbound connections...